src/components/TransactionBlock/PlrStakingV2TransactionBlock.tsx

```diff
@@ -299,6 +310,19 @@ const PlrStakingV2TransactionBlock = ({
     receiverAddress,
   ]);
 
+  const getBestRouteItem = (routes: Route[]) => {
+    let bestRoute = routes[0];
+    let minAmount = routes[0].gasCostUSD ? +routes[0].fromAmountUSD - +routes[0].gasCostUSD : Number.MAX_SAFE_INTEGER;
+
+    routes.forEach((route) => {
+      const { gasCostUSD, fromAmountUSD } = route;
+      if (!gasCostUSD) return;
+      if (+fromAmountUSD - +gasCostUSD > minAmount) bestRoute = route;
```

Comment:
should't it also check against current lowest `bestRoute`? because this loop would assign last conditionally acceptable `route` to `bestRoute` and would not compare against current best

Reply:
@poocart This should work right ?? When we get the bestRoute from the list, I am also updating the minAmount with that routes details.
  ```
const getBestRouteItem = (routes: Route[]) => {
    let bestRoute = routes[0];
    let minAmount = routes[0].gasCostUSD ? +routes[0].fromAmountUSD - +routes[0].gasCostUSD : Number.MAX_SAFE_INTEGER;

    routes.forEach((route) => {
      const { gasCostUSD, fromAmountUSD } = route;
      if (!gasCostUSD) return;
      if (+fromAmountUSD - +gasCostUSD > minAmount) {
        bestRoute = route;
        minAmount = +route.fromAmountUSD - +(route.gasCostUSD || 0);
      }
    });

    return bestRoute;
  };
```

Reply:
hmmm yes, but wondering about the equation being in minAmount favor, shouldn't it be that the equation always check for lower than mintAmount and not higher?

Reply:
@parag-chauhan

Reply:
@poocart When I changed the condition to this it is selecting below routes. Is this correct ??

```
if (+fromAmountUSD - +gasCostUSD < minAmount) {
   bestRoute = route;
   minAmount = +fromAmountUSD - +gasCostUSD;
}
```
![image](https://github.com/etherspot/etherspot-react-transaction-buidler/assets/109526304/6657e4cf-f5b1-44a6-a7a5-c1fff35e1d3c)

With this condition
```
if (+fromAmountUSD - +gasCostUSD >  minAmount) {
   bestRoute = route;
   minAmount = +fromAmountUSD - +gasCostUSD;
}
```
![image](https://github.com/etherspot/etherspot-react-transaction-buidler/assets/109526304/0ddbe5be-badb-4210-b5e5-c0741aaeac10)

Reply:
got it, it's ok then, `minAmount` is what to be received so then having it higher makes sense

Reply:
so then previous logic did make sense, correct?

Reply:
@parag-chauhan

Reply:
It looks like old logic makes sense.

Reply:
@poocart I also confirmed with Partha as well in the frontend channel so new logic makes sense to me now.